scripts/init-variables

```diff
@@ -14,9 +14,8 @@ AWS_REGION="$1"
 COREOS_CHANNEL="$2"
 COREOS_VM_TYPE="$3"
 
-COREOS_AMI_ID=`curl -s \
-  $(printf "http://%s.release.core-os.net/amd64-usr/current/coreos_production_ami_%s_%s.txt" \
-    $COREOS_CHANNEL $COREOS_VM_TYPE $AWS_REGION)`
+COREOS_AMI_ID=$(curl --silent https://coreos.com/dist/aws/aws-$COREOS_CHANNEL.json | grep -A 2 "$AWS_REGION" | grep $COREOS_VM_TYPE | cut -d"\"" -f4)
+[ -z "$COREOS_AMI_ID" ] && echo "FAIL: region $1 not found or doesn't include a CoreOS $COREOS_CHANNEL, $COREOS_VM_TYPE backed AMI" && exit 1
 
```

Comment:
Good catch, I suspect this is unnecessary now, turns out there was something wrong with the AMI (us-west-1, stable. hvm) that CoreOS providing in both the JSON or release.core-os.net.  I'll revert it.  I adapted the JSON parsing from here: https://fak3r.com/2015/07/09/howto-get-a-new-coreos-ami-id/  I mis-read the deprecation of the guy's code and jumping to conclusions about release.core-os.net